### ISSUE TITLE
Deleting a Particles2D object with a ParticlesMaterial causes an error

### ISSUE THREAD

**Issue by vybr** [godotengine/godot#14746]
**Operating system or device, Godot version, GPU Model and driver (if graphics related):**
3.0 Beta 1 - Windows

**Issue description:**
I have a projectile object, with a particle and sprite as children. **Queue_free()** is called on the projectile when it comes close to an enemy.
When the object is destroyed, this **[error appears in the debugger](https://imgur.com/IuqBw21)**.

`Type: Error`
`Description:  `
`Time: 0:00:03:0991  `
`C Error: Condition ' !id_map.has(p_rid.get_data())' is true. returned: 0  `
`C Source: c:\projects\godot-builds\godot\core\rid.h:164  `
`C Function: RID_Owner<struct RasterizerStorageGLES3::Material>::getornull`

 It doesn't matter if the particle object is being deleted directly or if it's the child of an object being deleted. The particle object is using the **ParticlesMaterial resource**. 

When I **turn emitting off, remove ParticlesMaterial or make the object invisible** , the error does not appear.

This essentially means I can't use particles unless I add a script to every particle object that removes the shader material when it exits the tree, which is just annoying.

EDIT: This also causes the game to crash when exported without the debugger. 
EDIT2: Also happens with ShaderMaterial.

**Comment by reduz** [comment]
The project fill form clearly asks to provide an example project. I can't reproduce, as this may be platform dependent (or I may be doing something wrong). 

Please provide one.

**Comment by vybr** [comment]
Apologies if my tone came across badly in that post. I appreciate the hard work guys put into this.

I didn't think an example was necessary as it literally happens every time queue_free is called on a Particles2D.
Just now I made a scene with a particles2D as the root node, added a built-in script:
```
extends Particles2D

_ready():
    queue_free()
```

I ran it with F6 and same happens.

Here's the example, which somehow produces 2 errors for me instead of 1:
[ParticlesBug.zip](https://github.com/godotengine/godot/files/1587528/ParticlesBug.zip)

**Comment by Dillybob1992** [comment]
Can confirm issue here. Windows 7, 64-bit, GTX 950. God forbid let's hope it's not yet another NVIDIA driver issue. @vybr What is your graphics card? This is really important

Although, I have not noticed any abnormal side effects with this error, but it is an *error*

Temporary solution (ty vybr), if anyone else stumbles upon this issue:
```
func _exit_tree():
	WeaponParticle2DNode.process_material = null
```

**Comment by brainsick** [comment]
Ubuntu 17.10, Godot 3.0 Beta 2, ParticlesBug.zip, can confirm the following errors:

```
ERROR: getornull: Condition ' !id_map.has(p_rid.get_data()) ' is true. returned: __null
   At: core/rid.h:164.
ERROR: getornull: Condition ' !id_map.has(p_rid.get_data()) ' is true. returned: __null
   At: core/rid.h:164.

```

**Comment by vybr** [comment]
@Dillybob1992 My PC uses a HD 6950, my laptop uses an Intel HD 4000. Forgot to mention that.

It crashes when exported (as with every error?).

**Comment by Dillybob1992** [comment]
Doesn't crash for me, might be specific crashing for AMD cards however

**Comment by vybr** [comment]
Ah, I just tried it on my PC (haven't had access to it until today) and the crash only happens on the Intel HD 4000 laptop.

**Comment by vybr** [comment]
This problem has been fixed by #15170
